src/serviceWorker/document.ts

```diff
@@ -0,0 +1,114 @@
+import { RouteHandlerCallbackOptions, RouteMatchCallbackOptions } from 'workbox-core'
+import { getCacheKeyForURL, matchPrecache } from 'workbox-precaching'
+import { Route } from 'workbox-routing'
+
+import { isDevelopment } from './utils'
+
+const fileExtensionRegexp = new RegExp('/[^/?]+\\.[^/]+$')
+export const DOCUMENT = process.env.PUBLIC_URL + '/index.html'
+
+/**
+ * Matches with App Shell-style routing, so that navigation requests are fulfilled with an index.html shell.
+ * See https://developers.google.com/web/fundamentals/architecture/app-shell
+ */
+export function matchDocument({ request, url }: RouteMatchCallbackOptions) {
+  // If this isn't a navigation, skip.
+  if (request.mode !== 'navigate') {
+    return false
+  }
+
+  // If this looks like a resource (ie has a file extension), skip.
+  if (url.pathname.match(fileExtensionRegexp)) {
+    return false
+  }
+
+  // If this isn't app.uniswap.org (or a local build), skip.
+  // IPFS gateways may not have domain separation, so they cannot use document caching.
+  if (url.hostname !== 'app.uniswap.org' && !isDevelopment()) {
```

Comment:
If we want to have service workers like they do, will have to mod this file

Reply:
I will create a separate task for this.
It will probably take some time to understand this I suggest we do this after the merge at some point